daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -243,6 +249,8 @@ class DataProcessor private (
 
   def withValidationMode(mode:ValidationMode.Type): DataProcessor = copy(validationMode = mode)
 
+  def withValidator(validator: Validator): DataProcessor = copy(validationMode = ValidationMode.Custom(validator))
```

Comment:
Convenience method and also illustrates that Validators are now passed without any other Config or SPI related data.

@stevedlawrence